packages/react-composites/tests/browser/chat/app/index.tsx

```diff
@@ -39,11 +40,34 @@ function App(): JSX.Element {
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, []);
 
-  return (
-    <>
-      {chatAdapter && <ChatComposite identifiers={IDS} adapter={chatAdapter} options={{ showParticipantPane: true }} />}
-    </>
-  );
+  if (customDataModel) {
+    return (
+      <>
+        {chatAdapter && (
```

Comment:
nit: much of this is consistent across the if statement regardless of customDataModel, extract what is applicable to the conditional to better contextualize why the if is needed and reduce redundancy
```
const onRenderTypingIndicator = customDataModel ? () => <text id="custom-data-model-typing-indicator">Someone is typing...</text> | undefined;
```

Reply:
I'd go a step further: The `customDataModel` URL parameter is problematic in two ways:
* It's not clear in `ChatComposite.test.ts` what setting this URL param does. When reading the test, I have to go see the implementation of `app/index.tsx` to get an idea of what the test does.
* Makes `app/index.tsx` inflexible and tied to that single test case.

Instead, how about adding URL parameters:
* `overrideTypingIndicator=<value>`
* `overrideRenderMessage=<value>`
* `overrideRenderAvatar=<value>`

And that test just sets all three:

    page.goto('....overrideRenderAvatar=Avatar&...')

Now, if we wanted to, we could split that test easily into three different cases -- one for each render override.

This also makes the implementation in `app.index.tsx` more natural:

    onRenderMessage={()=>{params.overrideRenderMessage ? <text id="overide-render-message">{params.overideRenderMessage}</text> : undefined}}

Reply:
The issue with that is that it only allows you to provide text values. We won't be able to test complex components as custom data model.

Reply:
> nit: much of this is consistent across the if statement regardless of customDataModel, extract what is applicable to the conditional to better contextualize why the if is needed and reduce redundancy
> 
> ```
> const onRenderTypingIndicator = customDataModel ? () => <text id="custom-data-model-typing-indicator">Someone is typing...</text> | undefined;
> ```

Can do.